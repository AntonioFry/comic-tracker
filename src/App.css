.rail-category {
  margin: 0px;
  padding: 0px 1% 0px;
  width: 98%;
}

.comic-rails {
  margin: 30px 0px;
}