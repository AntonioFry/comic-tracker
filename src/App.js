import React, { Component } from 'react';
import './App.css';
import { Header } from './Components/Header/Header';
import { ComicRail } from './Components/ComicRail/ComicRail';
import ComicDetails from './Components/ComicDetails/ComicDetails';
import { getWeeklyComics } from './API/apicalls';
import { connect } from 'react-redux';
import { setWeeklyComics } from './Actions/index';
import { Route } from 'react-router-dom';

class App extends Component {
  constructor() {
    super();
  }
  
  async componentDidMount() {
    try {
      const weeklyComics = await getWeeklyComics();
      await this.props.setWeeklyComics(weeklyComics);
    } catch (error) {
      console.log(error)
    }
  }

  getListOfAllComics = () => {
     if (this.props.comics === {}) {
       return
     } else {
       const allValues = Object.values(this.props.comics);
       const joinedValues = allValues.reduce((joined, collection) => {
         joined.push(...collection);
         return joined;
       }, [])
       return joinedValues;
     }
  }
  
  render() {
    const everyIssue = Array.from(new Set(this.getListOfAllComics()));
    const routesToIssues = everyIssue.map(comic => {
      return (
        <Route exact path={`/${comic.id}`} render={() => <ComicDetails 
          id={comic.id}
        />}/>
      )
    })
    return (
      <main>
        <Header />
        <Route exact path="/" render={() => (
          <section className="comic-rails">
            <h2 className="rail-category">This Weeks Comics</h2>
            {this.props.comics.weeklyComics ? <ComicRail comics={this.props.comics.weeklyComics} /> : null}
          </section>
        )} />
        {routesToIssues}
      </main>
    );
  }
}

const mapStateToProps = (store) => ({
  comics: store.comics
})

const mapDispatchToProps = (dispatch) => ({
  setWeeklyComics: comics => dispatch(setWeeklyComics(comics))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);