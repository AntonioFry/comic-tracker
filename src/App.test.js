import React from 'react';
import ReactDOM from 'react-dom'
import { App } from './App';
import { BrowserRouter } from 'react-router-dom';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';
import { cleanup, waitFor } from '@testing-library/react';


describe('App', () => {
  let comics;
  let characters;
  
  beforeEach(() => {
    characters = [
      {
        name: "Spider-Man",
        id: 1009610,
        desciption: "Bitten by a radioactive spider, high school student Peter Parker gained the speed, strength and powers of a spider. Adopting the name Spider-Man, Peter hoped to start a career using his new abilities. Taught that with great power comes great responsibility, Spidey has vowed to use his powers to help people."
      }
    ];
    comics = {
      "Weekly Comics": [],
      "Spider-Man Comics": [
        {
          title: "Marvel Age Spider-Man Vol. 2: Everyday Hero (Digest)",
          id: 1308,
          cover: { path: "http://i.annihil.us/u/prod/marvel/i/mg/9/20/4bc665483c3aa", extension: "jpg" }
        },
        {
          title: "Amazing Spider-Man (1999) #558 (Turner Variant)",
          id: 21171,
          cover: { path: "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available", extension: "jpg" }
        }
      ]
    };
  });

  afterEach(cleanup)

  it('Renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
    <BrowserRouter>
      <App characters={characters} comics={comics} />
    </BrowserRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });


  it('Matches snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App characters={characters} comics={comics} />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should call getWeeklyComics when component mounts', async () => {
    const div = document.createElement('div');

    const component = ReactDOM.render(
      <BrowserRouter>
        <App characters={characters} comics={comics} />
      </BrowserRouter>, div
    );

    
    await waitFor(() => (component.App.componentDidMount()));


    
  });

  it('Should return all comics in in one array when getListOfAllComics is called', () => {

  });

  it('Should change state when toggleNavBar is called', () => {

  });

})