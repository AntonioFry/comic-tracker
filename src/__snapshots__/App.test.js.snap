exports[`App Matches snapshot 1`] = `
<main>
  <div
    className={undefined}>
    <nav
      className="nav-bar-start">
      <img
        className="icon-start"
        onClick={[Function]}
        src="hamburger-icon.png" />
      <a
        className="nav-bar-links"
        href="/"
        onClick={[Function]}>
        HOME
      </a>
      <a
        className="nav-bar-links"
        href="/characters"
        onClick={[Function]}>
        CHARACTERS
      </a>
      <a
        className="nav-bar-links"
        href="/saved-comics"
        onClick={[Function]}>
        SAVED
      </a>
    </nav>
  </div>
  <header>
    <a
      className="logo-link"
      href="/"
      onClick={[Function]}>
      <img
        alt="Marvel comics logo"
        className="marvel-logo"
        src="marvel-logo.png" />
    </a>
  </header>
  <section
    className="comic-rails">
    <section
      className="comic-rail-container">
      <h2
        className="rail-category">
        Weekly Comics
      </h2>
      <div
        className="comic-rail">
        <h2
          className="no-comics-message">
          There are no comics available in the requested category
        </h2>
      </div>
    </section>
  </section>
</main>
`;
